npep/npep-122.md

```diff
@@ -0,0 +1,188 @@
+
+    As a cluster admin, I want to build tenants in my cluster that are isolated from each other by default. 
+    Tenancy may be modeled as 1:1, where 1 tenant is mapped to a single Namespace, or 1:n, where a single tenant 
+    may own more than 1 Namespace.
+
+The wording used here has lead to much confusion. Specifically, the "by default" part actually means this policy should 
+be at the BANP priority, and solves the "Engineering org vs Marketing org" use case where you want to keep orgs 
+from accidentally interfering with each other. By default, cross-tenant traffic should be dropped. 
+However, namespace owners can override this behavior by applying their own policies as needed. 
+
+In reality, the CR attached to this story defines strict tenancy, which should be a separate use case (see next section).
+
+To make the use case more obvious we can add some details to it like the following
+
+    As a cluster admin, I want to build tenants in my cluster that are isolated from each other by default.
+    By default, cross-tenant traffic is dropped. However, namespace owners can override this behavior by applying 
+    their own policies as needed.    
+    Tenancy may be modeled as 1:1, where 1 tenant is mapped to a single Namespace, or 1:n, where a single tenant 
+    may own more than 1 Namespace.
+
+#### Story 4.2: Create and Isolate multiple tenants in a cluster, strict isolation
+
+Strict tenancy is the "Coke vs Pepsi" sort of thing where you want each tenant to feel like it has its own cluster, 
+and be totally independent of the other tenants. We can write it down like this
+
+    As a cluster admin, I want to build tenants in my cluster that are isolated from each other where this isolation
+    can't be overridden by namespace owners. This policy should make every tenant completely independent and isolated 
+    from another tenants. Tenancy may be modeled as 1:1, where 1 tenant is mapped to a single Namespace, or 1:n, where a single tenant 
+    may own more than 1 Namespace.
+
+#### Story 4.3: Allow internal connections for tenants
+    
+    As a cluster admin, I want to build tenants in my cluster and always allow connections inside one tenant.
+    At the same time I want to setup an overridable deny-all policy to protect namespaces by default.
+    This policy should make sure internal connectivity for a tenant is always allowed, in case there are
+    lower-priority deny rules.
+
+#### Story 4.4: Tenants interaction with (B)ANP
+
+    As a cluster admin, I want to be able to setup policies with higher and lower priority than tenancy policy.
+    Higher priority to allow/deny some inter-tenant connections, for example, allow ingress traffic from the
+    monitoring namespace without including that namespace to every tenant. Lower priority to allow/deny
+    connections that are not a part of tenant policy, for example, allow internal tenant connections with tenancy
+    and deny-all policy with lower priority.
```

Comment:
The wording here is vague / too terse. I don't understand what you're saying.

Does "without including that namespace to every tenant" mean "without having to write a separate policy for every tenant"? If so, I'd suggest that wording.

I can't figure out what the lower priority example means at all.

Reply:
I tried to re-word it, ptal